File: Guanabara - POO Java/Meus exercícios POO/Aula10/src/main/java/heranca/Pessoa.java
package heranca;
public abstract class Pessoa { //`abstract class` não pode ser instanciada, assim não será possivel criar objetos a partir dela
    //Atributos
    private String nome;
    private int idade;
    private char sexo;
    
    //Método Público
    public final void fazerAniv(){ //Método Final não pode ser sobrescrito pelas suas subclasses
        this.idade++;
    }
    //Métodos Acessores
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade) {
        this.idade = idade;
    }

    public char getSexo() {
        return sexo;
    }

    public void setSexo(char sexo) {
        this.sexo = sexo;
    }
    
    //Método toString apresenta os dados da classe
    @Override
    public String toString() {
        return "Pessoa{" + "nome=" + nome + ", idade=" + idade + ", sexo=" + sexo + '}';
    }
}

File: Guanabara - PHP/exercicios-PHPOO/exercicios-PHPOO/Aula10-HerancaPessoa/Pessoa.php
<?php
abstract class Pessoa //Classe Abstrata não pode ser instanciada, ou seja, não é possível criar objetos a partir dela, só podendo servir como superclasse/progenitora.
{
    //Atributos
    protected $nome; //visibilidade protegida, ou seja, a classe mãe e suas filhas têm acesso ao atributo
    protected $idade;
    protected $sexo;

    //Métodos Especias
    public final function fazerAniversario() //Método Final não permite polimorfismo, ou seja, não pode ser sobrescrito por suas subclasses. Use para garantir que um determinado algoritmo não possa ser modificado pelas subclasses.
    {
        echo "<p><strong>" . $this->nome . "</strong> fez aniversário. Parabéns!</p>";
        $this->idade++;
    }

    //Métodos Acessores
    function getNome()
    {
        return $this->nome;
    }
    function getIdade()
    {
        return $this->idade;
    }
    function getSexo()
    {
        return $this->sexo;
    }
    function setNome($Nome)
    {
        $this->nome = $Nome;
    }
    function setIdade($Idade)
    {
        $this->idade = $Idade;
    }
    function setSexo($Sexo)
    {
        $this->sexo = $Sexo;
    }
}
